### ISSUE TITLE
How to display conditional widget depends on selected answer ?

### ISSUE THREAD

**Issue by MaheshPeri19** [cph-cachet/research.package#53]
Hi,

I want to display widget depends on user's answer. For example, question "Do you smoke ?", if user selection is Yes, I want to display one widget, if No, I want to display one widget.

Can you suggest me that can be achievable using this package ? If yes, please suggest me how to do it.

**Comment by MadsVSChristensen** [comment]
Hi @MaheshPeri19 

It is not possible to do as part of Research Package, but I have a possible workaround that I had success with.

Your RPUITask can have an `onSubmit` that navigates to different page like so (however, only after the survey is done) :
```
return RPUITask(
      task: task,
      onSubmit: (RPTaskResult result) {
        // if (result.results['<question identifier>'].results['answer'].first.value == <Value of the RPChoice>) {
        // Example below:
        if (result.results['smokingQuestionId'].results['answer'].first.value ==
            1) {
          Navigator.pushReplacement(context,
              MaterialPageRoute(builder: (context) => LinearSurveyPage()));
        }
      },
      ...
    );
```

But your task should then have `closeAfterFinish = false`

**Comment by MaheshPeri19** [comment]
Thank You. I will heck.

**Comment by MadsVSChristensen** [comment]
Closing this issue as a solution has been provided. If you are looking for a different type of solution, feel free to create a new issue on this. 

Mads
